packages/acceptance-tests/pkg-tests-specs/sources/commands/workspaces/foreach.test.js

```diff
@@ -192,6 +192,35 @@ describe(`Commands`, () => {
       )
     );
 
+    test(
+      `should not fall into endless loop if foreach run cmd is the same as lifecycle script name`,
+      makeTemporaryEnv(
+        {
+          private: true,
+          scripts: {
+            print: `yarn workspaces foreach run --parallel --topological --jobs=2 print`
```

Comment:
> Sounds good - one alternative would be to make `yarn workspaces foreach` only iterate on the children workspaces of the current worktree rather than all those in the project. How would you feel about it?
Yes, that would be better, why didn't I thought about this :)